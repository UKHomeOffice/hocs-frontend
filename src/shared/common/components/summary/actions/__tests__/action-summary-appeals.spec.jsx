import React from 'react';
import ActionSummaryAppeals from '../action-summary-appeals';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';

describe('action-summary-appeals.jsx', () => {

    it('should render one completed and one pending appeal', () => {

        const props = {
            items: [
                {
                    title: 'Appeal Type A',
                    outcome: null,
                    officerDirectorate: 'Directorate 2',
                    officer: 'Officer\'s Name',
                    status: 'No',
                    complexCase: null,
                    note: null,
                    dateSentRMS: null
                },{
                    title: 'Appeal Type B',
                    outcome: 'Decision upheld in part',
                    officerDirectorate: 'Directorate 2',
                    officer: 'Officer\'s Name',
                    status: 'Yes',
                    complexCase: 'Yes',
                    note: 'Some note detailing things',
                    dateSentRMS: '09/11/2021'
                },
            ]
        };

        expect(render(
            <>
                <ActionSummaryAppeals items={props.items} />
            </>
        )).toMatchSnapshot();
    });
});
