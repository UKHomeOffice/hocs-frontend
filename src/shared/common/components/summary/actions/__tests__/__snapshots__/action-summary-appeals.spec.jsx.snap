// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action-summary-appeals.jsx should render one completed and one pending appeal 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h2
        class="govuk-heading-m"
      >
        Appeals
      </h2>
      <table
        class="govuk-table margin-left--small"
      >
        <caption
          class="govuk-table__caption margin-bottom--small"
        >
          Appeal Type A
        </caption>
        <tbody
          class="govuk_table__body"
        >
          <tr
            class="govuk-table__row"
          >
            <th
              class="govuk-table__header padding-left--small govuk-!-width-one-third"
            >
              Directorate
            </th>
            <td
              class="govuk-table__cell"
            >
              Directorate 2
            </td>
          </tr>
          <tr
            class="govuk-table__row"
          >
            <th
              class="govuk-table__header padding-left--small govuk-!-width-one-third"
            >
              Officer Name
            </th>
            <td
              class="govuk-table__cell"
            >
              Officer's Name
            </td>
          </tr>
          <tr
            class="govuk-table__row"
          >
            <th
              class="govuk-table__header padding-left--small govuk-!-width-one-third"
            >
              Completed
            </th>
            <td
              class="govuk-table__cell"
            >
              No
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="govuk-table margin-left--small"
      >
        <caption
          class="govuk-table__caption margin-bottom--small"
        >
          Appeal Type B
        </caption>
        <tbody
          class="govuk_table__body"
        >
          <tr
            class="govuk-table__row"
          >
            <th
              class="govuk-table__header padding-left--small govuk-!-width-one-third"
            >
              Directorate
            </th>
            <td
              class="govuk-table__cell"
            >
              Directorate 2
            </td>
          </tr>
          <tr
            class="govuk-table__row"
          >
            <th
              class="govuk-table__header padding-left--small govuk-!-width-one-third"
            >
              Officer Name
            </th>
            <td
              class="govuk-table__cell"
            >
              Officer's Name
            </td>
          </tr>
          <tr
            class="govuk-table__row"
          >
            <th
              class="govuk-table__header padding-left--small govuk-!-width-one-third"
            >
              Completed
            </th>
            <td
              class="govuk-table__cell"
            >
              Yes
            </td>
          </tr>
          <tr
            class="govuk-table__row"
          >
            <th
              class="govuk-table__header padding-left--small govuk-!-width-one-third"
            >
              Completion date
            </th>
            <td
              class="govuk-table__cell"
            >
              09/11/2021
            </td>
          </tr>
          <tr
            class="govuk-table__row"
          >
            <th
              class="govuk-table__header padding-left--small govuk-!-width-one-third"
            >
              Outcome
            </th>
            <td
              class="govuk-table__cell"
            >
              Decision upheld in part
            </td>
          </tr>
          <tr
            class="govuk-table__row"
          >
            <th
              class="govuk-table__header padding-left--small govuk-!-width-one-third"
            >
              Complex case
            </th>
            <td
              class="govuk-table__cell"
            >
              Yes
            </td>
          </tr>
          <tr
            class="govuk-table__row"
          >
            <th
              class="govuk-table__header padding-left--small govuk-!-width-one-third"
            >
              Details
            </th>
            <td
              class="govuk-table__cell"
            >
              Some note detailing things
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <h2
      class="govuk-heading-m"
    >
      Appeals
    </h2>
    <table
      class="govuk-table margin-left--small"
    >
      <caption
        class="govuk-table__caption margin-bottom--small"
      >
        Appeal Type A
      </caption>
      <tbody
        class="govuk_table__body"
      >
        <tr
          class="govuk-table__row"
        >
          <th
            class="govuk-table__header padding-left--small govuk-!-width-one-third"
          >
            Directorate
          </th>
          <td
            class="govuk-table__cell"
          >
            Directorate 2
          </td>
        </tr>
        <tr
          class="govuk-table__row"
        >
          <th
            class="govuk-table__header padding-left--small govuk-!-width-one-third"
          >
            Officer Name
          </th>
          <td
            class="govuk-table__cell"
          >
            Officer's Name
          </td>
        </tr>
        <tr
          class="govuk-table__row"
        >
          <th
            class="govuk-table__header padding-left--small govuk-!-width-one-third"
          >
            Completed
          </th>
          <td
            class="govuk-table__cell"
          >
            No
          </td>
        </tr>
      </tbody>
    </table>
    <table
      class="govuk-table margin-left--small"
    >
      <caption
        class="govuk-table__caption margin-bottom--small"
      >
        Appeal Type B
      </caption>
      <tbody
        class="govuk_table__body"
      >
        <tr
          class="govuk-table__row"
        >
          <th
            class="govuk-table__header padding-left--small govuk-!-width-one-third"
          >
            Directorate
          </th>
          <td
            class="govuk-table__cell"
          >
            Directorate 2
          </td>
        </tr>
        <tr
          class="govuk-table__row"
        >
          <th
            class="govuk-table__header padding-left--small govuk-!-width-one-third"
          >
            Officer Name
          </th>
          <td
            class="govuk-table__cell"
          >
            Officer's Name
          </td>
        </tr>
        <tr
          class="govuk-table__row"
        >
          <th
            class="govuk-table__header padding-left--small govuk-!-width-one-third"
          >
            Completed
          </th>
          <td
            class="govuk-table__cell"
          >
            Yes
          </td>
        </tr>
        <tr
          class="govuk-table__row"
        >
          <th
            class="govuk-table__header padding-left--small govuk-!-width-one-third"
          >
            Completion date
          </th>
          <td
            class="govuk-table__cell"
          >
            09/11/2021
          </td>
        </tr>
        <tr
          class="govuk-table__row"
        >
          <th
            class="govuk-table__header padding-left--small govuk-!-width-one-third"
          >
            Outcome
          </th>
          <td
            class="govuk-table__cell"
          >
            Decision upheld in part
          </td>
        </tr>
        <tr
          class="govuk-table__row"
        >
          <th
            class="govuk-table__header padding-left--small govuk-!-width-one-third"
          >
            Complex case
          </th>
          <td
            class="govuk-table__cell"
          >
            Yes
          </td>
        </tr>
        <tr
          class="govuk-table__row"
        >
          <th
            class="govuk-table__header padding-left--small govuk-!-width-one-third"
          >
            Details
          </th>
          <td
            class="govuk-table__cell"
          >
            Some note detailing things
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
